### PR TITLE
DM-34599: Create AllSkyCam RubinTV channel

### PR DESCRIPTION
**Pull request by mfisherlevine**

### CODE REVIEW

File: python/lsst/rubintv/production/allSky.py [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+import subprocess
+
+from lsst.summit.utils.utils import (dayObsIntToString,
+                                     getCurrentDayObs_int,
+                                     getCurrentDayObs_datetime,
+                                     )
+from lsst.rubintv.production import Uploader
+from lsst.rubintv.production.rubinTv import _dataIdToFilename
+
+try:
```
Comment (czwa) [inline python/lsst/rubintv/production/allSky.py:36]: `HAS_GOOGLE_STORAGE` isn't used anywhere.

File: python/lsst/rubintv/production/allSky.py [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+    if not os.path.isdir(path):
+        raise RuntimeError(f"Cannot get files from {path}: it is not a directory")
+    files = [f for fname in os.listdir(path) if (os.path.isfile(f := os.path.join(path, fname)))]
+    files = [f for f in files if f.endswith(filetype)]
+    return set(files)
+
+
+class DayAnimator():
+    """A class for creating all sky camera stills and animations for a single
+    specificed given day.
```
Comment (czwa) [inline python/lsst/rubintv/production/allSky.py:161]: Typo.

File: python/lsst/rubintv/production/allSky.py [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+        The path to write the converted images out to. Need not exist, but must
+        be creatable with write privileges.
+    outputMovieDir : `str`
+        The path to write the movies. Need not exist, but must be creatable
+        with write privileges.
+    uploader : `lsst.rubintv.production.Uploader`
+        The uploader for sending images and movies to GCS.
+    channel : `str`
+        The name of the channel. Must match a channel name in rubinTv.py.
+    historical : `bool`
```
Comment (czwa) [inline python/lsst/rubintv/production/allSky.py:187]: optional

File: python/lsst/rubintv/production/allSky.py [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+        else:
+            seqNum = seqNumFromFilename(lastfile)
+            seqNumStr = f"{seqNum:05}"
+
+        channel = 'all_sky_movies'
+        fakeDataId = {'day_obs': self.dayObsInt, 'seq_num': seqNumStr}
+        uploadAsFilename = _dataIdToFilename(channel, fakeDataId, extension='.mp4')
+        creationFilename = os.path.join(self.outputMovieDir, uploadAsFilename)
+        self.log.info(f"Creating movie from {self.outputImageDir} as {creationFilename}...")
+        if not DRY_RUN:
```
Comment (czwa) [inline python/lsst/rubintv/production/allSky.py:283]: I think this is missing the `HAS_GOOGLE` check that I commented about above.

**Review by czwa — APPROVED** [review]

File: python/lsst/rubintv/production/allSky.py [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+    """
+    try:
+        os.makedirs(path, exist_ok=True)
+    except Exception as e:
+        raise RuntimeError(f'Error creating/accessing output path {path}') from e
+    if not os.access(path, os.W_OK):
+        raise RuntimeError(f"Output path {path} is not writable.")
+
+
+def dayObsFromDirName(fullDirName, logger):
```
[the remark this answers is not in the document]
Comment (mfisherlevine) [reply python/lsst/rubintv/production/allSky.py:69]: This is an interesting one. Not this, _per se_, but what's interesting is the fact that I have a notebook for regenerating all the historical movies, and that uses it, and so I'm not sure what I should do with this. Perhaps move it to the notebook, and stick the notebook in here in a notebooks dir?! Or is that too much informal cruft being put into the package - I really dont' know...

[the remark this answers is not in the document]
Comment (czwa) [reply python/lsst/rubintv/production/allSky.py:69]: If it's called elsewhere, then it's fine to stay here, but it probably should be added to `__all__`.  Noting that it's used there would be useful so it doesn't get caught in a review in the future.

File: python/lsst/rubintv/production/allSky.py [hunk 6]
```diff
@@ -0,0 +1,468 @@
...
+try:
+    from google.cloud import storage  # noqa: F401
+    HAS_GOOGLE_STORAGE = True
+except ImportError:
+    HAS_GOOGLE_STORAGE = False
+
+__all__ = ['DayAnimator', 'AllSkyMovieChannel']
+
+FPS = 10
+DRY_RUN = False
```
[the remark this answers is not in the document]
Comment (mfisherlevine) [reply python/lsst/rubintv/production/allSky.py:45]: Moved inside 👍

File: python/lsst/rubintv/production/allSky.py [hunk 7]
```diff
@@ -0,0 +1,468 @@
...
+        os.makedirs(path, exist_ok=True)
+    except Exception as e:
+        raise RuntimeError(f'Error creating/accessing output path {path}') from e
+    if not os.access(path, os.W_OK):
+        raise RuntimeError(f"Output path {path} is not writable.")
+
+
+def dayObsFromDirName(fullDirName, logger):
+    """Get the dayObs from the directory name.
+
```
[the remark this answers is not in the document]
Comment (mfisherlevine) [reply python/lsst/rubintv/production/allSky.py:71]: They have docstrings, but I just didn't bother with parameters stuff in places where it felt onerous and basically pointless. Plus for this one I was expecting the comment below and thought I might end up removing it 😄

[the remark this answers is not in the document]
Comment (mfisherlevine) [reply python/lsst/rubintv/production/allSky.py:71]: Will params all round though.

[the remark this answers is not in the document]
Comment (mfisherlevine) [reply python/lsst/rubintv/production/allSky.py:71]: OK, updated everywhere.

File: python/lsst/rubintv/production/allSky.py [hunk 8]
```diff
@@ -0,0 +1,468 @@
...
+        dirname = '20' + dirname
+        dayObsInt = int(dirname)
+        dayObsStr = dayObsIntToString(dayObsInt)
+        return dayObsInt, dayObsStr
+    except Exception:
+        logger.warning(f"Failed to parse directory name {fullDirName}")
+        return None, None
+
+
+def convertJpgScale(inFilename, outFilename):
```
[the remark this answers is not in the document]
Comment (mfisherlevine) [reply python/lsst/rubintv/production/allSky.py:91]: As discuss OOB, this is considered probably OK for now. Is about to be discussed in #software-dev, so I'll try to merge before that concludes, lest I have to change this! 😄

File: python/lsst/rubintv/production/allSky.py (hunk 1 above)
Comment (mfisherlevine) [reply python/lsst/rubintv/production/allSky.py:36]: Yup, fair. I was expecting to need a different `Uploader` but turns out the existing one was fine. Gone.

File: python/lsst/rubintv/production/allSky.py (hunk 4 above)
Comment (mfisherlevine) [reply python/lsst/rubintv/production/allSky.py:283]: That's OK, the whole init will fail elsewhere if that's not available, so I've removed all checks as they're done on instantiating the `Uploader`

**Pull request merged**